From e88870f2d5c212c2d11e78447cd2b2bdaa0fbc6b Mon Sep 17 00:00:00 2001
From: Alessio Ferri <alessio.ferri.3012@gmail.com>
Date: Wed, 17 Jun 2026 20:55:58 +0000
Subject: [PATCH 1/8] bcma: core: clear only FORCEHT for BCMA_CLKMODE_DYNAMIC

Clear FORCEHT with bcma_mask32() and leave the other CLKCTLST bits as
they are.
---
 drivers/bcma/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/bcma/core.c b/drivers/bcma/core.c
index 37a5ffe67..f4246689e 100644
--- a/drivers/bcma/core.c
+++ b/drivers/bcma/core.c
@@ -100,7 +100,7 @@ void bcma_core_set_clockmode(struct bcma_device *core,
 			bcma_err(core->bus, "HT force timeout\n");
 		break;
 	case BCMA_CLKMODE_DYNAMIC:
-		bcma_set32(core, BCMA_CLKCTLST, ~BCMA_CLKCTLST_FORCEHT);
+		bcma_mask32(core, BCMA_CLKCTLST, ~BCMA_CLKCTLST_FORCEHT);
 		break;
 	}
 }

base-commit: 24a19b953e2a3d0ea5a8c4951a0ab8f32720c907
